### PR TITLE
feat(pattern/tooltip): add tooltip pattern and documentation

### PR DESCRIPTION
**Pull request by mohamedMok**
## I have read [the contributing guidelines](https://mozaic.adeo.cloud/Contributing/)

- [x] Yes
- [ ] No

## Does this PR introduce a [breaking change](https://mozaic.adeo.cloud/Contributing/Developers/GitConventions/#breaking-changes-)?

- [ ] Yes
- [x] No

## Describe the changes

GitHub issue number or Jira issue URL: N/A

### CODE REVIEW

File: src/docs/Components/Tooltip/code.mdx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+@import 'settings-tools/_all-settings';
+
+@include import-font-families();
+
+@import 'components/_c.tooltip';
+```
+
+### Basic usage
+
+To create a basic tabs, wrap a link `a` inside a `ul` and apply the following classes:
```
Comment (nherwegh) [inline src/docs/Components/Tooltip/code.mdx:26]: a basic <s>tabs</s> tooltip

**Comment by tiloyi** [comment]
Deployment url : http://feattooltipsdev.demo.mozaic-design-system-dtrp.appspot.com/Components/Tooltip/

File: packages/styles/components/_c.tooltip.scss [hunk 2]
```diff
@@ -0,0 +1,107 @@
+/* mqp:start */
+.mc-tooltip {
+  $parent: &;
+
+  @include set-font-face();
+
+  position: relative;
+  display: inline-block;
+  margin: 0.25rem;
```
Comment (tiloyi) [inline packages/styles/components/_c.tooltip.scss:9]: I don't think that's necessary.

File: packages/styles/components/_c.tooltip.scss [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+  &__text {
+    visibility: hidden;
+    background-color: $color-grey-700;
+    color: $color-grey-200;
```
Comment (tiloyi) [inline packages/styles/components/_c.tooltip.scss:14]: I have the impression that this is not an accessible colour that can be used for text. @nherwegh is this voluntary?

File: packages/styles/components/_c.tooltip.scss [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    text-align: center;
+    border-radius: get-border-radius('m');
+    padding: 0.25rem 1rem;
+    position: absolute;
+    z-index: 1;
```
Comment (tiloyi) [inline packages/styles/components/_c.tooltip.scss:19]: I think we need to increase the value of the z-index. Maybe 1000?

File: packages/styles/components/_c.tooltip.scss [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+    &:focus,
+    &:hover {
+      #{$parent}__text {
+        visibility: visible;
+        left: 130%;
+
+        &::after {
+          top: 50%;
+          right: 100%;
+          margin-top: -4px;
```
Comment (tiloyi) [inline packages/styles/components/_c.tooltip.scss:40]: Replace by: 
```CSS
left: 0;
transform: translate(-100%, -50%)
```

File: packages/styles/components/_c.tooltip.scss [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+    &:focus,
+    &:hover {
+      #{$parent}__text {
+        visibility: visible;
+        right: 130%;
+
+        &::after {
+          top: 50%;
+          left: 100%; /* To the right of the tooltip */
+          margin-top: -5px;
```
Comment (tiloyi) [inline packages/styles/components/_c.tooltip.scss:58]: Replace by:
```CSS
right: 0;
transform: translate(100%, -50%);
```

File: packages/styles/components/_c.tooltip.scss [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+    &:hover {
+      #{$parent}__text {
+        visibility: visible;
+        left: 130%;
+
+        &::after {
+          top: 50%;
+          right: 100%;
+          margin-top: -4px;
+          border-width: 4px;
```
Comment (tiloyi) [inline packages/styles/components/_c.tooltip.scss:41]: ~~4px~~ => It is preferable to use the magic unit.

File: packages/styles/components/_c.tooltip.scss [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+    &:hover {
+      #{$parent}__text {
+        visibility: visible;
+        right: 130%;
+
+        &::after {
+          top: 50%;
+          left: 100%; /* To the right of the tooltip */
+          margin-top: -5px;
+          border-width: 5px;
```
Comment (tiloyi) [inline packages/styles/components/_c.tooltip.scss:59]: ~~5px~~ => It is preferable to use the magic unit.

File: packages/styles/components/_c.tooltip.scss [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+  }
+
+  &--top {
+    &:focus,
+    &:hover {
+      #{$parent}__text {
+        visibility: visible;
+        bottom: 130%;
+        left: 50%;
+        margin-left: -40px;
```
Comment (tiloyi) [inline packages/styles/components/_c.tooltip.scss:73]: ~~margin-left: -40px;~~ => `transform: translateX(-50%);`

File: packages/styles/components/_c.tooltip.scss [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+
+        &::after {
+          top: 100%; /* At the bottom of the tooltip */
+          left: 50%;
+          margin-left: -5px;
```
Comment (tiloyi) [inline packages/styles/components/_c.tooltip.scss:78]: Replace by

```CSS
bottom: 0;
left: 50%;
transform: translate(-50%, 100%);
```

File: packages/styles/components/_c.tooltip.scss [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+          border-width: 5px;
```
Comment (tiloyi) [inline packages/styles/components/_c.tooltip.scss:79]: ~~5px~~ => It is preferable to use the magic unit.

File: packages/styles/components/_c.tooltip.scss [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
+  }
+
+  &--bottom {
+    &:focus,
+    &:hover {
+      #{$parent}__text {
+        visibility: visible;
+        top: 130%;
+        left: 50%;
+        margin-left: -40px;
```
Comment (tiloyi) [inline packages/styles/components/_c.tooltip.scss:93]: ~~margin-left: -40px;~~ => `transform: translateX(-50%);`

File: packages/styles/components/_c.tooltip.scss [hunk 13]
```diff
@@ -0,0 +1,107 @@
...
+
+        &::after {
+          bottom: 100%;  /* At the top of the tooltip */
+          left: 50%;
+          margin-left: -5px;
```
Comment (tiloyi) [inline packages/styles/components/_c.tooltip.scss:98]: Replace by:
```CSS
top: 0;
left: 50%;
transform: translate(-50%, -100%);
```

File: packages/styles/components/_c.tooltip.scss [hunk 14]
```diff
@@ -0,0 +1,107 @@
...
+          border-width: 5px;
```
Comment (tiloyi) [inline packages/styles/components/_c.tooltip.scss:99]: ~~5px~~ => It is preferable to use the magic unit.

File: packages/styles/components/_c.tooltip.scss (hunk 3 above)
Comment (nherwegh) [reply packages/styles/components/_c.tooltip.scss:14]: Nope it's not. It's my mistake I gave `grey-200` for the font-color but it is: `$color-grey-000`

File: src/docs/Components/Tooltip/code.mdx [hunk 15]
```diff
@@ -0,0 +1,67 @@
...
+status:
+  sketch: 'bêta'
+  scss: 'bêta'
+---
+
+## Basic implementation
+
+### Import
+
+Import **the settings** and **the tabs** `scss` files.
```
Comment (nherwegh) [inline src/docs/Components/Tooltip/code.mdx:14]: the <s>tabs</s> tooltips …

File: src/docs/Components/Tooltip/code.mdx [hunk 16]
```diff
@@ -0,0 +1,67 @@
...
+
+@import 'components/_c.tooltip';
+```
+
+### Basic usage
+
+To create a basic tooltip, apply the following classes:
+
+- `mc-tooltip` on a div container
+- `mc-tabs__text` on the span tag
```
Comment (nherwegh) [inline src/docs/Components/Tooltip/code.mdx:29]: <s>tabs</s> tooltips

File: src/docs/Components/Tooltip/specs.mdx [hunk 17]
```diff
@@ -0,0 +1,28 @@
...
+| Variations                  | Height       | Paddings                  | Font-size             |
+| --------------------------- | ------------ | ------------------------- | --------------------- |
+| **Default**                 | 52px (3.25mu)| 16px (1mu)                | `size.font.04` (14px) |
+
+
+### Color
+
+| Variations                  | Background        | Text color                |
+| --------------------------- | ----------------- | ------------------------- |
+| **Default**                 | `$color-grey-700` | `$color-grey-200`         |
```
Comment (nherwegh) [inline src/docs/Components/Tooltip/specs.mdx:27]: Text color is: `$color-grey-000`

File: packages/styles/components/_c.tooltip.scss (hunk 2 above)
Comment (tiloyi) [reply packages/styles/components/_c.tooltip.scss]: You forgot to remove `display: inline-block;`

File: packages/styles/components/_c.tooltip.scss [hunk 18]
```diff
@@ -0,0 +1,110 @@
+/* mqp:start */
+.mc-tooltip {
+  $parent: &;
+
+  @include set-font-face();
+
+  position: relative;
+  display: inline-block;
+
```
Comment (tiloyi) [inline packages/styles/components/_c.tooltip.scss:9]: You should add : 

```CSS
&:focus,
&:hover {
  #{$parent}__text {
    visibility: visible;
  }
}
```

File: packages/styles/components/_c.tooltip.scss [hunk 19]
```diff
@@ -0,0 +1,110 @@
...
+      position: absolute;
+      border-style: solid;
+    }
+  }
+
+  &--right {
+    &:focus,
+    &:hover {
+      #{$parent}__text {
+        visibility: visible;
```
Comment (tiloyi) [inline packages/styles/components/_c.tooltip.scss:33]: To be removed : ~~visibility: visible;~~

File: packages/styles/components/_c.tooltip.scss [hunk 20]
```diff
@@ -0,0 +1,110 @@
...
+        }
+      }
+    }
+  }
+
+  &--left {
+    &:focus,
+    &:hover {
+      #{$parent}__text {
+        visibility: visible;
```
Comment (tiloyi) [inline packages/styles/components/_c.tooltip.scss:53]: To be removed : ~~visibility: visible;~~

File: packages/styles/components/_c.tooltip.scss [hunk 21]
```diff
@@ -0,0 +1,110 @@
...
+        }
+      }
+    }
+  }
+
+  &--top {
+    &:focus,
+    &:hover {
+      #{$parent}__text {
+        visibility: visible;
```
Comment (tiloyi) [inline packages/styles/components/_c.tooltip.scss:73]: To be removed : ~~visibility: visible;~~

File: packages/styles/components/_c.tooltip.scss [hunk 22]
```diff
@@ -0,0 +1,110 @@
...
+        }
+      }
+    }
+  }
+
+  &--bottom {
+    &:focus,
+    &:hover {
+      #{$parent}__text {
+        visibility: visible;
```
Comment (tiloyi) [inline packages/styles/components/_c.tooltip.scss:93]: To be removed : ~~visibility: visible;~~

File: src/docs/Components/Tooltip/index.mdx [hunk 23]
```diff
@@ -9,7 +9,7 @@ status:
 
 > A tooltip allows you to display further informations to the user when needed. It can also be used to explicitely specify a visual element like an icon.
 
---preview of the four tooltips--
```
Comment (caqueste) [inline src/docs/Components/Tooltip/index.mdx:12]: Just a little detail → 2px are missing for the tooltip height.
Can you improve the arrow like that we have in Sketch?

tooltip-arrow

Comment (tiloyi) [reply src/docs/Components/Tooltip/index.mdx]: @caqueste Your remarks have been taken into account. It's been corrected.

**Pull request merged**